src/hooks/swap/useRouterUrl.tsx

```diff
@@ -0,0 +1,11 @@
+import { createContext, PropsWithChildren, useContext } from 'react'
+
+const Context = createContext<string | undefined>(undefined)
+
+export function RouterUrlProvider({ children, routerUrl }: PropsWithChildren<{ routerUrl?: string }>) {
```

Comment:
we already have the `swapRouterUrlAtom` so you could use that instead of doing the context

seems more consistent bcuz we store other main props in atoms ^